### ISSUE TITLE
Implementing `pthread_key`

### ISSUE THREAD

**Issue by carbotaniuman** [sunfishcode/mustang#131]
In my journey to get `bat` working, I'm now stuck on `pthread_key`. The semantics of it seem pretty simple as I just need a counter and a fixed-size array on the stack for each thread.

I do see stuff in origin for handling this, but I'm not sure where to add the code and what else needs to be changed.

**Comment by sunfishcode** [comment]
We could add this to origin, but the `pthread_key` API is fairly C-specific; newer languages and systems have tended towards `thread_local` keywords/macros. So my preference is to implement this entirely in c-scape, rather than add this to origin.

In c-scape, we can write something like:

```rust
#[thread_local]
static VALUES: RefCell<[*mut c_void; PTHREAD_KEYS_MAX]> = RefCell::new([null_mut(); PTHREAD_KEYS_MAX]);
```
to allocate the arrays for this. Then it's a matter of adding the metadata for allocating them via `pthread_key_create`, and calling their destructors. Origin has `at_thread_exit` which we can use here, though it doesn't have any support for *unregistering* a function once it's registered, so c-scape should register its own function that then calls all the active destructors, rather than registering all the destructors individually.

**Comment by sunfishcode** [comment]
This is implemented in #132!
